File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(const char* cards1[], size_t cards1_len, const char* cards2[], size_t cards2_len, const char* goal[], size_t goal_len) {
    int idx1 = 0;
    int idx2 = 0;
    
    for(int i = 0; i < goal_len; i++){
        if(idx1 < cards1_len && strcmp(cards1[idx1], goal[i]) == 0){
            idx1++;
        }else if(idx2 < cards2_len && strcmp(cards2[idx2], goal[i]) == 0){
            idx2++;
        }else{
            return "No";
        }
    }
    
    return "Yes";
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer = 'Yes'
    
    for i in range(len(goal)):
        if not cards1 and not cards2:
            break;
        if cards1 and cards1[0] == goal[i]:
            cards1.pop(0)
        elif cards2 and cards2[0] == goal[i]:
            cards2.pop(0)
        else:
            return 'No'
            
    return answer